zfs/zfs.go

```diff
@@ -618,11 +638,21 @@ func (n *NilBool) String() string {
 	return fmt.Sprintf("%v", n.B)
 }
 
+func (n *NilBool) IsTrue() bool {
```

Comment:
Sry, can't have that!
The reason for `NilBool`'s existence is to have a type that doesn't default to something:

```
type Foo struct {
  ImportantConfigKnob *NilBool
}

...

func (f Foo) {
   if f.ImportantConfigKnob.B {
   
   } else {
   }
}
```

If someone declares such a struct, they express the intent that they always want it be set explicitly, and that they would rather crash than have the implicit default value there (because the caller forgot to set it explicitly).
(Default values are a double-edged sword, and for important features such as *should we send the user data encrypted or unencrypted*, I'd much rather crash if we forgot to set that field somewhere, than do the wrong thing).

---

Feel free to add the above as a doc comment on `NilBool`.

---

As to your usage of `NilBool`: I think most of the send options (`LargeBlock`, `EmbeddedData`, etc. can safely default to `false`). Thus you could just use a normal `bool`. But if you think the caller should be punished for forgetting to set the fields, leave them as `NilBool`s. You just can't have `IsTrue()`, it defeats the purpose of `NilBool`.

Reply:
Ah, thanks! While writing this I was already wondering whether this is the right thing to do.